### PR TITLE
Fix flakiness in TestDataTemplateGeneratorCmdLineApp

### PR DESCRIPTION
**Pull request by evanw555**
The temp folder operations have unexpected behavior on some macOS
systems (e.g. "/var/folders", "/private/var/folders")

### CODE REVIEW

**Review by nickibi — APPROVED** [review]

File: generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java [hunk 1]
```diff
@@ -178,11 +178,17 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
       Assert.assertTrue(generatedSource.contains("class " + pegasusTypeName),
           "Incorrect generated class name.");
 
-      // Match the generated annotation (use regex since this is error-prone)
+      // Validate the generated annotation (use a strange regex-based comparison since temp files can be weird e.g. /var vs. /private/var)
```
Comment (junchuanwang) [inline generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java:181]: Users have been building in Mac from long time ago.
why is this previously not caught?

File: generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java [hunk 2]
```diff
@@ -178,11 +178,17 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
...
       final Matcher generatedAnnotationMatcher = GENERATED_ANNOTATION_PATTERN.matcher(generatedSource);
       Assert.assertTrue(generatedAnnotationMatcher.find(), "Unable to find a valid @Generated annotation in the generated source file.");
-      final String expectedGeneratedAnnotation = "Generated from " + expectedTypeNamesToSourceFileMap.get(pegasusTypeName);
-      Assert.assertEquals(generatedAnnotationMatcher.group(1), expectedGeneratedAnnotation, "Unexpected @Generated annotation.");
+      final String expectedPath = expectedTypeNamesToSourceFileMap.get(pegasusTypeName);
+      final String actualPath = generatedAnnotationMatcher.group(1);
```
Comment (junchuanwang) [inline generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java:185]: can we give comment examples? Although I know you can be pretty good at Regular Expression, so might be everyone else 

But natural language comments are still appreciated.

File: generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java (hunk 1 above)
Comment (evanw555) [reply generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java:181]: 1. This test was added recently.
2. It's possible that only some Mac systems have this temp folder quirk, as it doesn't seem to happen for all Mac builds.

File: generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java (hunk 2 above)
Comment (evanw555) [reply generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java:185]: Good point, this testing logic is a little convoluted. I'll update it

**Review by junchuanwang — APPROVED** [review]

**Pull request merged**